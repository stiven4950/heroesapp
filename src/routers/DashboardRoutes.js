import React from 'react';
import { Redirect, Route, Switch } from 'react-router';
import DcScreen from '../components/dc/DcScreen';
import HeroesScreen from '../components/heroes/HeroesScreen';
import MarvelScreen from '../components/marvel/MarvelScreen';
import SearchScreen from '../components/search/SearchScreen';
import Navbar from '../components/ui/Navbar';

const DashboardRoutes = () => {
    return (
        <>
            <Navbar />

            <div className='container mt-5'>
                <Switch>
                    <Route exact path='/marvel' component={MarvelScreen} />
                    <Route exact path='/heroe/:heroeId' component={HeroesScreen} />
                    <Route exact path='/dc' component={DcScreen} />
                    <Route exact path='/search' component={SearchScreen} />

                    {/* Si alguna de las rutas no se encuentra se hace el redireccionamiento */}
                    <Redirect to='/marvel' />
                </Switch>
            </div>
        </>
    );
}

export default DashboardRoutes;