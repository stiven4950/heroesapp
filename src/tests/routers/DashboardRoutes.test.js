import React from 'react';
import { mount } from 'enzyme';

import { AuthContext } from '../../auth/AuthContext';
import DashboardRoutes from '../../routers/DashboardRoutes';
import { MemoryRouter } from 'react-router';

describe('Pruebas en el componente <DashboardRoutes />', () => {
    test('Debe de mostrarse correctamente', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: {
                name: 'Omar',
                logged: true,
            }
        };
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Omar');
    });

});