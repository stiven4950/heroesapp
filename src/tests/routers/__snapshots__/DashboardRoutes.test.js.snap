// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <DashboardRoutes /> Debe de mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Omar
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-5"
  >
    Array [
      <h1>
        Marvel Screen
      </h1>,
      <hr />,
      <div
        className="row row-cols-1 row-cols-md-3 g-2 animate__animated animate__fadeIn"
      >
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 340,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Spider Man"
                className="card-img"
                src="./assets/heroes/marvel-spider.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Spider Man
                </div>
                <p
                  className="card-text"
                >
                  Peter Parker
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Amazing Fantasy #15
                  </small>
                </p>
                <a
                  href="/heroe/marvel-spider"
                  onClick={[Function]}
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 340,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Captain America"
                className="card-img"
                src="./assets/heroes/marvel-captain.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Captain America
                </div>
                <p
                  className="card-text"
                >
                  Steve Rogers
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Captain America Comics #1
                  </small>
                </p>
                <a
                  href="/heroe/marvel-captain"
                  onClick={[Function]}
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 340,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Iron Man"
                className="card-img"
                src="./assets/heroes/marvel-iron.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Iron Man
                </div>
                <p
                  className="card-text"
                >
                  Tony Stark
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Tales of Suspense #39
                  </small>
                </p>
                <a
                  href="/heroe/marvel-iron"
                  onClick={[Function]}
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 340,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Thor"
                className="card-img"
                src="./assets/heroes/marvel-thor.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Thor
                </div>
                <p
                  className="card-text"
                >
                  Thor Odinson
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Journey into Myster #83
                  </small>
                </p>
                <a
                  href="/heroe/marvel-thor"
                  onClick={[Function]}
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 340,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Hulk"
                className="card-img"
                src="./assets/heroes/marvel-hulk.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Hulk
                </div>
                <p
                  className="card-text"
                >
                  Bruce Banner
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    The Incredible Hulk #1
                  </small>
                </p>
                <a
                  href="/heroe/marvel-hulk"
                  onClick={[Function]}
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 340,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Wolverine"
                className="card-img"
                src="./assets/heroes/marvel-wolverine.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Wolverine
                </div>
                <p
                  className="card-text"
                >
                  James Howlett
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    The Incredible Hulk #180
                  </small>
                </p>
                <a
                  href="/heroe/marvel-wolverine"
                  onClick={[Function]}
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 340,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Daredevil"
                className="card-img"
                src="./assets/heroes/marvel-daredevil.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Daredevil
                </div>
                <p
                  className="card-text"
                >
                  Matthew Michael Murdock
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Daredevil #1
                  </small>
                </p>
                <a
                  href="/heroe/marvel-daredevil"
                  onClick={[Function]}
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 340,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Hawkeye"
                className="card-img"
                src="./assets/heroes/marvel-hawkeye.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Hawkeye
                </div>
                <p
                  className="card-text"
                >
                  Clinton Francis Barton
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    Tales of Suspense #57
                  </small>
                </p>
                <a
                  href="/heroe/marvel-hawkeye"
                  onClick={[Function]}
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 340,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Cyclops"
                className="card-img"
                src="./assets/heroes/marvel-cyclops.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Cyclops
                </div>
                <p
                  className="card-text"
                >
                  Scott Summers
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    X-Men #1
                  </small>
                </p>
                <a
                  href="/heroe/marvel-cyclops"
                  onClick={[Function]}
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card ms-3"
          style={
            Object {
              "maxWidth": 340,
            }
          }
        >
          <div
            className="row no-gutters"
          >
            <div
              className="col-md-4"
            >
              <img
                alt="Silver Surfer"
                className="card-img"
                src="./assets/heroes/marvel-silver.jpg"
              />
            </div>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Silver Surfer
                </div>
                <p
                  className="card-text"
                >
                  Norrin Radd
                </p>
                <p
                  className="card-text"
                >
                  <small
                    className="text-muted"
                  >
                    The Fantastic Four #48
                  </small>
                </p>
                <a
                  href="/heroe/marvel-silver"
                  onClick={[Function]}
                >
                  Más...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>,
    ]
  </div>,
]
`;
